spec/system/admin/order_cycles/simple_spec.rb

```diff
@@ -368,6 +368,47 @@
         end
       end
 
+      context "when variants are hidden via inventory settings" do
+        let(:oc) do
+          create(:simple_order_cycle, suppliers: [supplier_managed],
+                                      coordinator: distributor_managed,
+                                      distributors: [distributor_managed],
+                                      name: 'Order Cycle 1' )
+        end
+        let(:product) { create(:product, supplier: supplier_managed) }
+        let(:v1) { create(:variant, product: product ) }
+        let(:inventory_item_v1) {
+          create(:inventory_item, enterprise: distributor_managed, variant: v1, visible: false)
+        }
+        before do
+          # Incoming exchange
+          ex_in = oc.exchanges.where(sender_id: supplier_managed, receiver_id: distributor_managed,
+                                     incoming: true).first
+          ex_in.update(variant_ids: [v1.id])
+          ex_in.save!
+
+          # Outgoing exchange
+          ex_out = oc.exchanges.where(sender_id: distributor_managed,
+                                      receiver_id: distributor_managed, incoming: false).first
+          ex_out.update(variant_ids: [v1.id,])
+          v1.inventory_items = [inventory_item_v1]
+          ex_out.save!
+
+          # hide via inventory settings variant v1
+          supplier_managed.update preferred_product_selection_from_inventory_only: true
```

Comment:
I realize now, that this might not be the relevant setting: inventory is a feature relevant for hubs/shops and not suppliers. We should replace this with:

`distributor_managed.update preferred_product_selection_from_inventory_only: true`

And add a test where this setting is set to `false`, from which we expect the warning not to be displayed.